samples: add shell (#15)
NordicSemiconductor/npmx-zephyr: samples/shell/README.rst

.. _shell_sample:

SHELL
#####

.. contents::
   :local:
   :depth: 2

> The SHELL sample allows to use the Zephyr shell to control a PMIC device.

Author: Zephyr Shell driver

> The Shell sample allows to control a PMIC device with command set using the Zephyr shell module.

Author: Yes

> The Shell sample allows to control a PMIC device with command set using the Zephyr shell module.

Reviewer: No, the command set is built using the shell module.

> The Shell sample allows to control a PMIC device with a command set and using the Zephyr shell module.

Reviewer 2: ```suggestion
The Shell sample allows to control a PMIC device using a command set and the Zephyr shell module.
```

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

> This sample does the following:

Reviewer 2: ```suggestion
This sample allows to do the following:
```

> This sample do the following:
> 
> * Configure the Zephyr shell driver.
> * Initialize and register a POF callback.

Reviewer: ```suggestion
This sample does the following:

* Configures the Zephyr shell driver.
* Initializes and registers a POF callback.
```
What do you think @annakielar ?

Reviewer 2: Ohhh my bad. Definitely yes.

Author: FIxed

> * Initialize and register a POF callback.

Author: What about `The Shell sample allows to control a PMIC device with the Zephyr Shell driver.`?

        - P1.10
      * - GPIO1
        - P1.11
        - P1.11
      * - SDA
        - P1.02
        - P0.26
      * - SCL
        - P1.03
        - P0.27
      * - VOUT2 & GND
        - External supply (P21)
        - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * Connect a suitable battery to the **J2** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P2** pin header, connect **VBAT** and **VBATIN** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **RSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.

#. Make the following connections on the DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

.. |sample path| replace:: :file:`samples/shell`

.. include:: /includes/build_and_run.txt

Testing
=======

|test_sample|

#. |connect_kit|

> #. |connect_terminal|

Reviewer 2: Any sample-specific steps?

      * - GPIO1
        - P1.11
        - P1.11
      * - SDA
        - P1.02
        - P0.26
      * - SCL
        - P1.03
        - P0.27
      * - VOUT2 & GND
        - External supply (P21)
        - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * Connect a suitable battery to the **J2** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P2** pin header, connect **VBAT** and **VBATIN** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **RSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **RSET2** and **VSET2** pins with a jumper.

#. Make the following connections on the DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

.. |sample path| replace:: :file:`samples/shell`

.. include:: /includes/build_and_run.txt

Testing
=======

|test_sample|

#. |connect_kit|
#. |connect_terminal|

> #. Review and use npmx commands in the Shell.

Reviewer 2: ```suggestion
#. Review available npmx commands, and use the relevant ones in the shell.
```
